test/functional/wallet_balance.py

```diff
@@ -296,5 +303,27 @@ def test_balances(*, fee_node_1=0):
             assert_equal(self.nodes[0].getbalances()['mine']['untrusted_pending'], Decimal('0.1'))
 
 
+        # Tests the lastprocessedblock JSON object in getbalances, getwalletinfo
+        # and gettransaction by checking for valid hex strings and by comparing
+        # the hashes & heights between generated blocks.
+        self.log.info("Test getbalances returns expected lastprocessedblock json object")
+        prev_hash = self.nodes[0].getbestblockhash()
+        prev_height = self.nodes[0].getblock(prev_hash)['height']
+        self.generatetoaddress(self.nodes[0], 5, self.nodes[0].get_deterministic_priv_key().address)
+        lastblock = self.nodes[0].getbalances()['lastprocessedblock']
+        assert_is_hash_string(lastblock['hash'])
+        assert_equal((prev_hash == lastblock['hash']), False)
```

Comment:
there's also `assert_not_equal` that could be used

Reply:
`assert_not_equal` does not exist :neutral_face:

Reply:
right, sorry, mind was mixed up while reviewing: https://github.com/bitcoin/bitcoin/pull/23127